components/org.wso2.micro.integrator.core/src/main/java/org/wso2/micro/integrator/core/util/MicroIntegratorBaseUtils.java

```diff
@@ -560,4 +564,21 @@ public static void setCarbonAxisConfigurator(CarbonAxisConfigurator carbonAxisCo
     public static CarbonAxisConfigurator getCarbonAxisConfigurator() {
         return carbonAxisConfigurator;
     }
+
+    /**
+     * Get Synapse Environment.
+     *
+     * @return SynapseEnvironment - SynapseEnvironment
+     */
+    public static SynapseEnvironment getSynapseEnvironment() {
+
+        Parameter synapseEnvironmentParatemer =
+                CarbonCoreDataHolder.getInstance().getAxis2ConfigurationContextService().getServerConfigContext()
+                        .getAxisConfiguration().getParameter(SynapseConstants.SYNAPSE_ENV);
+        if (synapseEnvironmentParatemer != null) {
```

Comment:
Can this ever be null? Is this method getting called before the sunapseEnvironment is set?

Reply:
yes , since the synapse environment is initialized finally and this is in the core. Also we call this method only once and the null check won't be costly.

Reply:
but can the method 'getSynapseEnvironment' be called before the synapseEnvironment comes? It's not about the cost, this will hide erroneous situations where the synapseEnvironment was not set correctly.

Reply:
In coordination code the flow is as follows, 

Register Task Service -> Register Synapse Environment -> Deploy Tasks -> Then it will call this method to get Synapse Environment. ( So null issue won't be there ).

But since this method is in BaseUtils -> If someone uses it not properly there will be NPE. 
I preferred returning null instead of running into NPE in runtime. wdyt . Else we can add a comment saying that this will return NPE if called before initializing synapse environment. wdyt

Reply:
We need to have NPE if someone uses it not properly because that is a design failure. Otherwise this will return null, and we might have done null checks somewhere else at it will fail at a lower stage without giving us no clue about the synapseEnvironment being null.